individual/schema.py

```diff
@@ -44,18 +65,52 @@ def resolve_individual_data_source(self, info, **kwargs):
         if client_mutation_id:
             filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))
 
-        Query._check_permissions(info.context.user)
+        Query._check_permissions(info.context.user,
+                                 IndividualConfig.gql_individual_search_perms)
         query = IndividualDataSource.objects.filter(*filters)
         return gql_optimizer.query(query, info)
 
+    def resolve_group(self, info, **kwargs):
+        Query._check_permissions(
+            info.context.user,
+            IndividualConfig.gql_group_search_perms
+        )
+        filters = append_validity_filter(**kwargs)
+        client_mutation_id = kwargs.get("client_mutation_id", None)
+        if client_mutation_id:
+            filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))
+
+        query = Group.objects.filter(*filters)
+        return gql_optimizer.query(query, info)
+
+    def resolve_group_individual(self, info, **kwargs):
+        Query._check_permissions(
+            info.context.user,
+            IndividualConfig.gql_group_search_perms
+        )
+        filters = append_validity_filter(**kwargs)
+        client_mutation_id = kwargs.get("client_mutation_id", None)
+        if client_mutation_id:
+            filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))
+
+        query = GroupIndividual.objects.filter(*filters)
+        return gql_optimizer.query(query, info)
+
     @staticmethod
-    def _check_permissions(user):
-        if type(user) is AnonymousUser or not user.id or not user.has_perms(
-                IndividualConfig.gql_individual_search_perms):
+    def _check_permissions(user, perms):
+        if type(user) is AnonymousUser or not user.id or not user.has_perms(perms):
             raise PermissionError("Unauthorized")
 
 
 class Mutation(graphene.ObjectType):
     create_individual = CreateIndividualMutation.Field()
     update_individual = UpdateIndividualMutation.Field()
     delete_individual = DeleteIndividualMutation.Field()
+
+    create_group = CreateGroupMutation.Field()
+    update_group = UpdateGroupMutation.Field()
+    delete_group = DeleteGroupMutation.Field()
+
+    create_group_individual = CreateGroupIndividualMutation.Field()
+    update_group_individual = UpdateGroupIndividualMutation.Field()
+    delete_group_individual = DeleteGroupIndividualMutation.Field()
```

Comment:
I think those mutations could be named "add_individual_to_group", "edit_individual_in_group" and "remove_individual_from_group"

Reply:
done